openedx/openedx-platform
# Don't use OLX when copying/duplicating blocks [FC-0049]

## Description

Refactors the block/library copy/paste and duplicate functions to ensure that content tags are copy/pasted/duplicated to use signals/events to trigger copying tags instead of using the OLX as intermediate storage.

## Supporting information

Closes https://github.com/openedx/modular-learning/issues/203

## Testing instructions

USE latest master:
1. Create a course with units that contain all the different types of blocks, including a library content block.
1. Add tags to your units, and to the actual blocks.
2. Copy the tagged unit to staged content clipboard.

USE this branch:
1. Run migrations
2. Paste the block you copied into your course.

   Ensure the paste works without errors, and creates the expected content.
   Note that no tags will be pasted though.
3. Copy your tagged unit again.
4. Paste the tagged unit.

   Check that the tags pasted as expected in the parent and all the different types of blocks.
5. Duplicate a subsection/unit.
   Check that the tags duplicated as expected in the parent and child blocks.

## Deadline

Need to get this merged before Redwood is cut to avoid OLX incompatibilities in exported courses or staged content.

Private-ref: [FAL-3688](https://tasks.opencraft.com/browse/FAL-3688)
